src/hal/AngleSensor.h

```diff
@@ -3,12 +3,16 @@
 
 #include "SensorInterface.h"
 
-class AngleSensor : public SensorInterface {
+// int is just a placeholder
+class AngleSensor : public SensorInterface<double, int> {
 public:
     explicit AngleSensor(double initialAngle);
-    double GetAngle() override;
+    double GetValue(int gpio) override;
 
 private:
+    double ValidateValue(double angle) override;
+    static const double kMinAngle;
+    static const double kMaxAngle;
```

Comment:
No need to be `static`, you defined as `const` on .cpp file. Which makes sense to be just a constant and not a member of the class. So you can remove from here.